File: ios/BFExplorer/Plaoc/bfs/ipc/IpcStreamData.swift
//
//  IpcStreamData.swift
//  BFExplorer
//
//  Created by ui08 on 2023/2/8.
//

import Foundation

struct IpcStreamData {
    var type: IPC_DATA_TYPE = IPC_DATA_TYPE.stream_data
    let stream_id: String
    let data: Any
    
    init(stream_id: String, data: Any) {
        self.stream_id = stream_id
        self.data = data
    }
    
    static func fromBinary(ipc: Ipc, stream_id: String, data: Data) -> IpcStreamData {
        if ipc.support_message_pack {
            return IpcStreamData(stream_id: stream_id, data: data)
        }
        
        return IpcStreamData(stream_id: stream_id, data: data.base64EncodedString())
    }
}

extension IpcStreamData: IpcMessage {}

File: android/app/src/main/java/info/bagen/rust/plaoc/microService/ipc/IpcStreamData.kt
package info.bagen.rust.plaoc.microService.ipc

import info.bagen.rust.plaoc.microService.ipc.helper.IPC_DATA_TYPE
import info.bagen.rust.plaoc.microService.ipc.helper.SimpleEncoding
import info.bagen.rust.plaoc.microService.ipc.helper.simpleDecoder

/** 处理ipc流数据*/
class IpcStreamData(val stream_id: String, val data: Any) {
  val  type = IPC_DATA_TYPE.STREAM_DATA;
    /** ipc流数据转化为二进制*/
   fun fromBinary(ipc: Ipc, stream_id: String, data: ByteArray): IpcStreamData {
        if (ipc.supportMessagePack) {
            return  IpcStreamData(stream_id, data);
        }
        return  IpcStreamData(stream_id, simpleDecoder(data, SimpleEncoding.base64));
    }
}